### PR TITLE
Integrate components and fix bugs for EKS Container Insights

### PR DESCRIPTION
**Pull request by pxaws**
**Description:** <Describe what has changed.>
This PR integrates various components of Container Insights and fixes some bugs.
* set region in aws session for ec2 tags
* add decoration logic with host info
* add configurations for metric decoration with pod/service store
* initialize k8s client on demand to save memory
* fix the jitter issue that happens randomly
* add attributes to cluster-level metrics

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/2307

### CODE REVIEW

File: receiver/awscontainerinsightreceiver/internal/stores/podstore.go
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/stores/podstore.go:103]: Storing a `context.Context` in a struct is generally discouraged.  Can you do this another way?  If not, document here why this is necessary?

File: receiver/awscontainerinsightreceiver/internal/stores/podstore.go
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/stores/podstore.go:249]: What is the timeout on this `Context`?  If it is stored on the `PodStore` struct and not refreshed is it likely to be already expired or so long as to be ineffective?

File: receiver/awscontainerinsightreceiver/internal/stores/podstore.go
Comment (pxaws) [reply receiver/awscontainerinsightreceiver/internal/stores/podstore.go]: I moved the context to `store.go` and still save it in the struct (because I need to use the context when calling the Decorate function). I added some explanation there as well.

File: receiver/awscontainerinsightreceiver/internal/stores/podstore.go
Comment (pxaws) [reply receiver/awscontainerinsightreceiver/internal/stores/podstore.go]: There is no timeout. The context is defined in https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/da6ba4f5515a959883ab34430e2341893b52143a/receiver/awscontainerinsightreceiver/receiver.go#L69.

To make sure the context never timeout or cancel by itself, I changed it to use a background context
```
ctx, acir.cancel = context.WithCancel(context.Background())
```

File: internal/aws/k8s/k8sclient/endpoint.go
Comment (pxaws) [reply internal/aws/k8s/k8sclient/endpoint.go:53]: I have tried to implement the interface outside of this package. But golang doesn't allow me to do that. That's why I have to split it into two interfaces. I also don't want to expose the `shutdown()` outside the package because I want to centralize the control in K8sClient.

File: receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go:89]:
```suggestion
func WithDecorator(d decorator) Option {
```

It is idiomatic to name `Option` creation functions that provide a configuration value `WithX`.

File: receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go
Comment (Aneurysm9) [inline receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go:106]:
```suggestion
type Decorator interface {
```

Since this interface consists of exported functions and types, and it appears to be implemented outside of this package, it probably makes sense to export the interface as well to allow implementers to validate their implementation using this pattern:

```go
var _ cadvisor.Decorator = DecoratorImpl{}
```

Comment (pxaws) [reply receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go:106]: Good suggestion!

File: receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go
Comment (pxaws) [reply receiver/awscontainerinsightreceiver/internal/cadvisor/cadvisor_linux.go:89]: I will make the change.

**Review by Aneurysm9 — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]
Thanks

**Comment by alolita** [comment]
@open-telemetry/collector-approvers @jpkrohling can one of you please review before I mark this PR as ready-to-merge.

**Review by anuraaga — APPROVED** [review]

**Pull request merged**
